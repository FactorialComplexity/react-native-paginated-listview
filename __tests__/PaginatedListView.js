const { Text, ListView } = require('react-native');
const React = require('react');
const { shallow } = require('enzyme');
const renderer = require('react-test-renderer');

const PaginatedListView = require('../PaginatedListView');
const oneThroughTen = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];

describe('<PaginatedListView />', () => {
  it('Can render like a normal ListView', () => {
    const tree = renderer.create(
      <PaginatedListView
        renderRow={(data) => <Text>{data}</Text>}
        initialData={oneThroughTen}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();

    const wrapper = shallow(
      <PaginatedListView
        renderRow={(data) => <Text>{data}</Text>}
        initialData={oneThroughTen}
      />
    );
    
    expect(wrapper.length).toEqual(1);
    expect(wrapper.find(ListView).length).toEqual(1);
  });
});
