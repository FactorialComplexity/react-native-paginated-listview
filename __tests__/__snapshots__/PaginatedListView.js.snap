// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaginatedListView /> Can render like a normal ListView 1`] = `
<View>
  <RCTScrollView>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        1
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        2
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        3
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        4
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        5
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        6
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        7
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        8
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        9
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        10
      </Text>
    </View>
  </RCTScrollView>
</View>
`;
